pkg/store/bucket.go

```diff
@@ -1050,14 +1050,16 @@ func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Ag
 	hasher := hashPool.Get().(hash.Hash64)
 	defer hashPool.Put(hasher)
 
-	if in.Encoding() == chunkenc.EncXOR {
+	if in.Encoding() == chunkenc.EncXOR || in.Encoding() == chunkenc.EncHistogram {
 		b, err := save(in.Bytes())
 		if err != nil {
 			return err
 		}
-		out.Raw = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b, Hash: hashChunk(hasher, b, calculateChecksum)}
+		storeEnc := storepb.Chunk_Encoding(in.Encoding() - 1)
```

Comment:
I think we generally prefer to avoid variables used just one: https://thanos.io/v0.30/contributing/coding-style-guide.md/#avoid-defining-variables-used-only-once. I suggest moving this into the next line and split it over a few lines instead.

Reply:
Makes sense, updated.